### PR TITLE
Redaction shared

### LINKED ISSUE #45: Support deleting messages from the Swing GUI
When in a chat room and the user wants to delete a message
Then allow the user to delete a message.

### PR DESCRIPTION
**Pull request by enoxone**
About Time

### CODE REVIEW

File: serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt [hunk 1]
```diff
@@ -448,12 +471,17 @@ class MatrixSession(val client: HttpClient, val server: String, val user: String
             ?: "<no room name - $id>"
     }
     fun mergeInSync(new_sync_response: SyncResponse) {
+                        /*for(event in room.timeline.events)
```
Comment (lotusronin) [inline serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt]: Delete this comment block

File: serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt [hunk 2]
```diff
@@ -448,12 +471,17 @@ class MatrixSession(val client: HttpClient, val server: String, val user: String
...
+                            if(event is RoomMessageEvent && event.redacts != null) {
+                                getRedactEvent(room_id,event.redacts)
+                            }*/
```
Comment (Limvot) [inline serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt]: Comment block should be removed

File: serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt [hunk 3]
```diff
@@ -462,15 +490,30 @@ class MatrixSession(val client: HttpClient, val server: String, val user: String
...
+                            }
                         } catch (e: Exception) {
                             println("while trying to insert (from mergeInSync) ${event} got exception $e")
-
                         }
                     }
                 }
+                for (redactId in redactedEvents) {
+                    val freshRedact = getRedactEvent(room_id,redactId)
+                    println(freshRedact.toString())
```
Comment (Limvot) [inline serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt]: Probably don't need to keep this println

File: serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt [hunk 4]
```diff
@@ -462,15 +490,30 @@ class MatrixSession(val client: HttpClient, val server: String, val user: String
                             // Note that if this is the first event of this chunk, we record prev_batch for it
                             // so we can later backfill by picking the non-null prev_batch with smallest seqId
                             // for this room.
-                            println("adding event from sync at $insertId")
-                            Database.addRoomEvent(insertId, session_id, room_id, event, getRelatedEvent(event), if (index == 0 && room.timeline.limited) { room.timeline.prev_batch } else { null })
-                            insertId = idBetween(insertId, "z", true)
+                            if(event is RoomMessageEvent && event.redacts != null){
+                                println("this is a redaction")
```
Comment (lotusronin) [inline serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt]: Remove this println, the later for loop has a loop where it prints out the latest redacted event.

**Review by Limvot — APPROVED** [review]
Looks good over all, 2 nits

**Review by lotusronin — CHANGES_REQUESTED** [review]
Minor change requests, otherwise lgtm!

**Review by lotusronin — APPROVED** [review]
lgtm!

**Pull request merged**